packages/core/src/plugins/html-deserializer/utils/htmlTextNodeToString.ts

```diff
@@ -3,9 +3,101 @@
  */
 import { isHtmlText } from './isHtmlText';
 
-export const htmlTextNodeToString = (node: HTMLElement | ChildNode) => {
+function getStyleFromNode(node: HTMLElement | ChildNode): string {
+  if (!(node as HTMLElement).getAttribute) return '';
+
+  const styleAttr = (node as HTMLElement).getAttribute('style');
+
+  return styleAttr ? styleAttr.trim() : '';
+}
+
+// Can use the npm package style-to-object.
+// https://www.npmjs.com/package/style-to-object
+function styleToObject(styleText: string): Record<string, string> {
+  const styleObject = {};
+  const declarations = styleText.split(';');
+
+  const camelCase = (str: string) => {
+    return str.replaceAll(/-([a-z])/g, (match: string, letter: string) => {
+      return letter.toUpperCase();
+    });
+  };
+
+  declarations.forEach((declaration) => {
+    declaration = declaration.trim();
+    if (declaration) {
+      const [property, value] = declaration.split(':');
+      const propertyName = camelCase(property.trim());
+      // @ts-ignore
+      styleObject[propertyName] = value.trim();
+    }
+  });
+
+  return styleObject;
+}
+
+function findParentElementWhiteSpace(node: HTMLElement | ChildNode) {
+  let parentNode = node.parentNode;
+
+  while (parentNode != null) {
+    const styleStr = getStyleFromNode(parentNode as typeof node);
+    const styles = styleStr ? styleToObject(styleStr) : {};
+
+    if (parentNode.nodeType === Node.ELEMENT_NODE && styles['whiteSpace']) {
+      // If parentNode.nodeType === 'PRE', do not perform any processing here.
+      return styles['whiteSpace'];
```

Comment:
For the <pre> tag, if it needs to be handled, it should be addressed elsewhere. My suggestion is not to process the content within the <pre> tag. The expected behavior of the <pre> tag is already satisfactory.

Reply:
Do we need to handle it when it is set to "initial"?

Reply:
Parsing whitespace within the `<pre>` tag works correctly when Plate's code block plugin is enabled, but not otherwise. You can test this by pressing the "Customize" button on the playground and unchecking "Code block".

If users are pasting HTML content containing a `<pre>` tag into an editor that doesn't support code block, I think they would still expect the line breaks to be preserved.

Reply:
> Do we need to handle it when it is set to "initial"?

Yes, nice catch. :+1:

Reply:
I apologize for missing the implementation of the `<pre>` tag. It does not conflict with a code block.

Reply:
No worries. You were right to question whether it was (or should be) within the spec. 🙂